GVFS/GVFS.Common/ProductUpgraderInfoImpl.cs

```diff
@@ -0,0 +1,26 @@
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System;
+using System.IO;
+
+namespace GVFS.Common
+{
+    public abstract class ProductUpgraderInfoImpl

```

Comment:
I think the eventual goal here is to combine this and the `ProductUpgraderInfo` classes (or maybe this should just be a subclass of that type)...